### DISCUSSION TITLE
UnoCSS with Next.js

### DISCUSSION THREAD

**Discussion by derkoe** [vercel/next.js#33254] [Help]
Has anybody tried https://github.com/antfu/unocss with Next? Any best practices for integration?

UnoCSS seems like a better and more flexible alternative to Tailwind CSS - for an introduction see https://antfu.me/posts/reimagine-atomic-css

**Comment by 0xsven** [comment]
I got it working with UnoCSS' CLI:

install:
```
npm install unocss @unocss/cli -D
```

add to package json
```
"scripts": {
    "unocss:dev": "unocss \"./**/*.tsx\" --watch",
    "unocss:build": "unocss \"./**/*.tsx\""
}
```

add `unocss.config.js` file to root:
```
import { defineConfig, presetMini } from 'unocss';

export default defineConfig({
  presets: [presetMini()],
});
```

**Comment by ekil1100** [comment]
How to use it via webpack? The @unocss/webpack plugin is not working.

**Reply by AYEKYIPHYU92** [reply]
can you show me next.config.js

**Comment by seognil** [comment]
It's working (with Webpack cache off, otherwise it only works on the first build)

I create a repo as a demo with some explanation <https://github.com/seognil/nextjs-unocss-webpack>

---

`next.config.js`

```js
const UnoCSS = require("@unocss/webpack").default;
const presetUno = require("@unocss/preset-uno").default;

/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,

  webpack(config, context) {
    config.plugins.push(UnoCSS({ presets: [presetUno()] }));

    if (context.buildId !== "development") {
      // * disable filesystem cache for build
      // * https://github.com/unocss/unocss/issues/419
      // * https://webpack.js.org/configuration/cache/
      config.cache = false;
    }

    return config;
  },
};

module.exports = nextConfig;
```

`_app.js`

```tsx
import "uno.css";
```

`index.tsx`

```tsx
// * somewhere
<div className='text-red'><div>
```

**Reply by paulm17** [reply]
Unfortunately this doesn't work now.  Tried with latest nextjs and unocss.

**Reply by paulm17** [reply]
So I cloned the repo and it wouldn't install with pnpm.  Used yarn instead.  But it works.  🤔 

Edit: It's because the repo is pages and I'm using the appdir.

Ok, well that clarifies it for me.  Sticking with tailwind.

**Comment by JeDTr** [comment]
I found an examlpe here and it works fine https://github.com/unocss/unocss/blob/main/examples/next/README.md

**Reply by paulm17** [reply]
Sadly it's postcss and not lightningcss.  There doesn't seem an easy way to get these two running.

**Reply by millette** [reply]
Could <https://www.npmjs.com/package/postcss-lightningcss> help?

**Reply by paulm17** [reply]
@millette thanks for taking the time to respond, appreciate it 😄.

It probably could.  But I'd rather just use lightningcss and not have postcss in my stack.  

Tailwind with one of the experimental versions has support for lightning and will complete the migration to it, with v4.  

So I consider postcss to be deprecated.

**Comment by lveillard** [comment]
Hey, anyone trying this? 
Maybe using the runtime version? https://unocss.dev/integrations/runtime
